src/query/plan/pretty_print.cpp

```diff
@@ -589,6 +599,16 @@ bool PlanToJsonVisitor::PreVisit(Filter &op) {
   op.input_->Accept(*this);
   self["input"] = PopOutput();
 
+  auto i = 1;
+  for (const auto &pattern_filter : op.pattern_filters_) {
+    auto s = std::to_string(i);
+
+    pattern_filter->Accept(*this);
+    self["pattern_filter" + s] = PopOutput();
+
+    i++;
+  }
```

Comment:
The idea to make this more understandable:

```
for (const auto& pattern_filter : op.pattern_filters_) {
  const auto index = std::distance(op.pattern_filters_.begin(), std::find(op.pattern_filters_.begin(), op.pattern_filters_.end(), pattern_filter)) + 1;
  const auto key = "pattern_filter" + std::to_string(index);

  pattern_filter->Accept(*this);
  self[key] = PopOutput();
}
```

If not that, I would isolate "pattern_filter" + s as `key` variable

Reply:
```
  for (auto i = 0; i < op.pattern_filters_.size(); i++) {
    auto pattern_filter_key = "pattern_filter" + std::to_string(i);

    op.pattern_filters_[i]->Accept(*this);
    self[pattern_filter_key] = PopOutput();
  }

```

I went with this